### PR TITLE
adding switch button to switch Dynamic and static between planets. ad…

### PR DESCRIPTION
**Pull request by vaslee**
adding a switch button to switch Dynamic and static between planets. adding the rotating gestures

### CODE REVIEW

**Comment by vhart** [comment]
Hey @vaslee mind elaborating on the objective of this PR, like what it means to switch between dynamic and static? and perhaps linking to a trello card with the expected behavior?

File: StellarApp/Controllers/SolarViewController.swift [hunk 1]
```diff
@@ -20,20 +25,57 @@ class SolarViewController: UIViewController, ARSCNViewDelegate {
...
+            let rotateGestureRecognizer = UIRotationGestureRecognizer(target: self, action: #selector(rotate))
+            sceneView.addGestureRecognizer(tapGestureRecognizer)
+            sceneView.addGestureRecognizer(pinchGestureRecognizer)
+            sceneView.addGestureRecognizer(rotateGestureRecognizer)
+            
+            sceneView.scene.rootNode.addChildNode(sunNode)
+            Planet.getPlanets().forEach { sunNode.addChildNode($0) }
+            sceneView.isUserInteractionEnabled = true
+            
+        case .animation:
```
Comment (vhart) [inline StellarApp/Controllers/SolarViewController.swift]: this seems backwards. should animation remove all gestures?

File: StellarApp/Controllers/SolarViewController.swift [hunk 2]
```diff
@@ -20,20 +25,57 @@ class SolarViewController: UIViewController, ARSCNViewDelegate {
...
+            sceneView.scene.rootNode.enumerateChildNodes { (node, stop ) in
+                node.removeFromParentNode()
+            }
+            let tapGestureRecognizer = UITapGestureRecognizer(target: self, action: #selector(tapped))
+            let pinchGestureRecognizer = UIPinchGestureRecognizer(target: self, action: #selector(pinch))
+            let rotateGestureRecognizer = UIRotationGestureRecognizer(target: self, action: #selector(rotate))
+            sceneView.removeGestureRecognizer(tapGestureRecognizer)
+            sceneView.removeGestureRecognizer(pinchGestureRecognizer)
+            sceneView.removeGestureRecognizer(rotateGestureRecognizer)
```
Comment (vhart) [inline StellarApp/Controllers/SolarViewController.swift]: does rotating and panning have to be tied together? Is it possible that you may want to turn off rotation but leave on scaling/tapping? Seems like rotation is the only actual animation.

File: StellarApp/Controllers/SolarViewController.swift [hunk 3]
```diff
@@ -59,6 +101,18 @@ class SolarViewController: UIViewController, ARSCNViewDelegate {
...
     }
     
+    @objc func rotate(rotate: UIRotationGestureRecognizer) {
+       
+        let rotateView = rotate.view as! SCNView
+        let rotateLocation = rotate.location(in: rotateView)
+        let hitTest = rotateView.hitTest(rotateLocation, options: nil)
+        if !hitTest.isEmpty {
+            let rotateAction = SCNAction.rotateBy(x: 1, y: 1, z: 0, duration: 0.1)
+            SCNNode.deepScaleNode(node: sunNode, scaleAction: rotateAction)
```
Comment (vhart) [inline StellarApp/Controllers/SolarViewController.swift]: this is a rotate action not a scale action.

**Comment by vaslee** [comment]
because I adding the rotating gesture to the planets, but when I rotate during the planets is moving around, it's going mess up, all the orbits crossing to each other. That's why I adding the switch button, so it switches back and forth. when planets are moving, there are not gestures happen until user switch to another mode, which is planets not moving around. And this is not on the trello board yet because I don't know how to describe it.

**Comment by vaslee** [comment]
@vhart

**Comment by vhart** [comment]
Thanks for the explanation @vaslee. I'm happy to help you scope it in a ticket if you need help!

**Pull request merged**